pkg/utils/pod/terminating.go

```diff
@@ -0,0 +1,15 @@
+package pod
+
+import (
+	"time"
+
+	"github.com/awslabs/karpenter/pkg/utils/ptr"
+	v1 "k8s.io/api/core/v1"
+)
+
+func IsStuckTerminating(pod *v1.Pod) bool {
```

Comment:
I'd leave this scoped to the termination logic since I'd hope we wouldn't use this anywhere else.